JS/Lesson19/gulcin/recursion.js

```diff
@@ -123,10 +170,65 @@ const nestedObjEx7 = {
 	},
 	h: 5,
 };
+
+function findMaxDepthOfObj(obj) {
+  const entries = Object.entries(obj);
+  const keys = Object.keys(obj);
+  const restOfEntries = entries.slice(1);
+  const newOb = Object.fromEntries(restOfEntries);
+  const depth = 1;
+  if(keys.length === 0) {
+	return depth;
+  } else {
+      if(typeof obj[keys[0]] !== 'object') {
+        return findMaxDepthOfObj(newOb);
+     } 
+      if(typeof obj[keys[0]] === 'object') {
+        return 1 + findMaxDepthOfObj(obj[keys[0]]);
+     }
+  }
+}
+
+console.log(findMaxDepthOfObj(nestedObjEx7));
 // 8. Write a recursive function to reverse the order of words in a sentence.
 const sentenceEx8 = 'Hello, how are you?';
+
+function reverseWords(sentence) {
+	const wordsArr = sentence.split(' ');
+	if(wordsArr.length === 0 || wordsArr[0].length === 0 ) {
+		return '';
+	} else {
+	    let output = [];
+		output.push(wordsArr[wordsArr.length - 1]);
+		let restOfTheSentence = wordsArr.slice(0, wordsArr.length-1).join(' ');
+		return output.join('') + ' ' + reverseWords(restOfTheSentence);
+		
+	}
+}
+console.log(reverseWords(sentenceEx8));
 // 9. Write a recursive function to find the length of the longest word in a sentence.
 const sentenceEx9 = 'The quick brown fox jumps over the lazy dog';
+const sentenceEx99 = 'Can you give me information about it?';
+
+function longestWord(sentence) {
+  const sentenceArr = sentence.split(' ');
+  if(sentenceArr.length === 0 || sentenceArr[0].length === 0) {
+	return '';
+  } else {
+	const lengths = sentenceArr.map(word => word.length);
+	let restOfTheSentence = sentenceArr.slice(1).join(' ');
```

Comment:
`restOfTheSentence` does not change so it's better to use `const`
```suggestion
	const restOfTheSentence = sentenceArr.slice(1).join(' ');
```